### PR TITLE
Relax `pip` check in setup.py to allow installation via other package managers (e.g. `poetry`)

### PR DESCRIPTION
**Pull request by dsmilkov**
Remove redundant pip check which causes installation failure when using non-pip package managers like `poetry` or `pipenv`.  See https://github.com/python-poetry/poetry/issues/4888 for the issue the `pip` check causes in `poetry`.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by Abacn** [comment]
agree this code is far outdated. 
On the other side there was recently an issue of pypi dependency backoff with gcp dependencies and got mitigated. Maybe we still want some warning but update the recommended version number and phrasing, and also make sure this does not fail when there is no pypi (add None check or so). @tvalentyn may have more context about this.

File: sdks/python/setup.py [hunk 1]
```diff
@@ -95,16 +95,6 @@ def get_version():
 different technologies and user communities.
 '''
 
-REQUIRED_PIP_VERSION = '7.0.0'
```
Comment (Abacn) [inline sdks/python/setup.py:98]: The context was here: https://github.com/apache/beam/issues/22218#issuecomment-1196014115. We may change REQUIRED_PIP_VERSION -> RECOMMENDED_MIN_PIP_VERSION and change the warning accordingly. And put the code in a try clause and catch `DistributionNotFound`.

Comment (dsmilkov) [reply sdks/python/setup.py:98]: That sounds better and more robust. Done.

**Review by dsmilkov — COMMENTED** [review]
Done.

**Review by Abacn — APPROVED** [review]
LGTM

File: sdks/python/setup.py (hunk 1 above)
Comment (Abacn) [reply sdks/python/setup.py:98]: Thanks!

**Comment by Abacn** [comment]
R: @tvalentyn

**Comment by tvalentyn** [comment]
LGTM, we can merge when tests pass.

**Comment by tvalentyn** [comment]
please ping the PR if not merged tomorrow.

**Comment by dsmilkov** [comment]
Ping @tvalentyn @Abacn . Tests pass

**Comment by tvalentyn** [comment]
Thanks!

**Pull request merged**